airflow/models/dag.py

```diff
@@ -1394,21 +1422,30 @@ def handle_callback(self, dagrun, success=True, reason=None, session=NEW_SESSION
         :param reason: Completion reason
         :param session: Database session
         """
-        callbacks = self.on_success_callback if success else self.on_failure_callback
+        callbacks, context = (
+            DAG.fetch_callback(dag=self, dagrun=dagrun, success=success, reason=reason, session=session)
+            or None,
```

Comment:
Would be a good idea to put the function call on its own line, this comma (and lack of) is too easy to read incorrectly.